src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexReplacement.cs

```diff
@@ -35,7 +35,8 @@ public RegexReplacement(string rep, RegexNode concat, Hashtable _caps)
             if (concat.Type() != RegexNode.Concatenate)
                 throw new ArgumentException(SR.ReplacementError);
 
-            StringBuilder sb = StringBuilderCache.Acquire();
+            Span<char> buffer = stackalloc char[ReplaceMaxBufferSize];
```

Comment:
I believe that we typically use inline constants for these. Having a named constant that just used in a single place like this does not have a lot of value.